setup.sh

```diff
@@ -92,8 +93,20 @@ function check_dependencies()
     else
       eval "sudo $cmd"
     fi
-  fi
 
+    # installation failed...
+    local ret="$?"
+    if [[ "$ret" -ne 0 ]]; then
+      local errmsg=''
+      errmsg="[ERROR]:Dependency installation failure: "
+      [[ "$ret" -eq 1 ]] && errmsg+="lacked sudo privileges to install the packages "
+      [[ "$ret" -ne 1 ]] && errmsg+="something went wrong when installing "
```

Comment:
Use single-quotes in all these 3 lines. Also, don't forget to double check if these return codes are consistent, if you didn't already.